db.js

```diff
@@ -554,21 +558,22 @@ exports.init = function (sbot, config) {
       write(record) {
         const buf = record.value
         const pValue = buf ? bipf.seekKey(buf, 0, B_VALUE) : -1
-        indexesArr.forEach((idx) => idx.onRecord(record, false, pValue))
+        for (const idx of indexesArr) idx.onRecord(record, false, pValue)
       },
       end() {
         debug(`updateIndexes() scan time: ${Date.now() - start}ms`)
-        const flushTasks = indexesArr.map((idx) =>
-          promisify(idx.flush.bind(idx))()
-        )
-        Promise.all(flushTasks).then(() => {
+        const doneFlushing = multicb({ pluck: 1 })
+        for (const idx of indexesArr) idx.flush(doneFlushing())
+        doneFlushing((err) => {
```

Comment:
As a bonus, now we're handling errors from flushing indexes, while before we weren't.